unmarshal.go

```diff
@@ -93,24 +94,105 @@ func (f fqnToFields) setFeature(fqn string, value any) error {
 }
 
 func (result *OnlineQueryResult) unmarshal(resultHolder any) (returnErr *ClientError) {
-	fqnToValueMap := map[Fqn]any{}
+	fqnToValue := map[Fqn]any{}
 	for _, featureResult := range result.Data {
-		fqnToValueMap[featureResult.Field] = featureResult.Value
+		fqnToValue[featureResult.Field] = featureResult.Value
 	}
-	return UnmarshalInto(resultHolder, fqnToValueMap, result.expectedOutputs)
+	return UnmarshalInto(resultHolder, fqnToValue, result.expectedOutputs)
 }
 
-func UnmarshalInto(resultHolder any, fqnToValueMap map[Fqn]any, expectedOutputs []string) (returnErr *ClientError) {
+func unmarshalTableInto(table arrow.Table, resultHolders any) (returnErr error) {
+	defer func() {
+		if panicContents := recover(); panicContents != nil {
+			detail := "details irretrievable"
+			switch typedContents := panicContents.(type) {
+			case *reflect.ValueError:
+				detail = typedContents.Error()
+			case string:
+				detail = typedContents
+			}
+			returnErr = fmt.Errorf("exception occurred while unmarshalling result: %s", detail)
+		}
+	}()
+
+	slicePtr := reflect.ValueOf(resultHolders)
+	if slicePtr.Kind() != reflect.Ptr {
+		return fmt.Errorf(
+			"result holder should be a pointer to a slice of structs, "+
+				"got '%s' instead",
+			slicePtr.Kind(),
+		)
+	}
+
+	slice := reflect.Indirect(slicePtr)
+	if slice.Kind() != reflect.Slice && slice.Kind() != reflect.Array {
```

Comment:
The code below `append`s into the slice, so the `reflect.Array` case won't actually work, and shouldn't be permitted here.